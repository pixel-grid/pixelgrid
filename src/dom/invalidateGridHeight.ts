import getDocumentHeight from '../helpers/getDocumentHeight';

/**
 * Invalidate grid element height
 * @param element Grid's root element
 */
const invalidateGridHeight: (element: HTMLElement) => HTMLElement = (
    element: HTMLElement
) => {
    // Hide grid before measuring document height
    element.style.display = 'none';

    // Measure document height
    const height = getDocumentHeight();

    // Update Grid's height and show
    element.style.height = `${height}px`;
    element.style.display = 'block';

    return element;
};

export default invalidateGridHeight;
