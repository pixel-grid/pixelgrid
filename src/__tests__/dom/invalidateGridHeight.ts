jest.mock('../../helpers/getDocumentHeight', () => ({
    __esModule: true,
    default: () => 500
}));

import invalidateGridHeight from '../../dom/invalidateGridHeight';

describe('invalidateGridHeight', () => {
    test('returns correct element', () => {
        const sourceElement = document.createElement('div');

        const result = invalidateGridHeight(sourceElement);

        expect(result).toEqual(sourceElement);
    });

    test("invalidates element's height", () => {
        const sourceElement = document.createElement('div');

        invalidateGridHeight(sourceElement);

        expect(sourceElement).toMatchSnapshot();
    });

    test("invalidates element's height for grid larger than document", () => {
        const sourceElement = document.createElement('div');
        sourceElement.style.height = '100000px';

        invalidateGridHeight(sourceElement);

        expect(sourceElement).toMatchSnapshot();
    });
});
