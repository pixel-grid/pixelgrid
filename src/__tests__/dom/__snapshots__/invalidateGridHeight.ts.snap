// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalidateGridHeight invalidates element's height 1`] = `
<div
  style="display: block; height: 500px;"
/>
`;

exports[`invalidateGridHeight invalidates element's height for grid larger than document 1`] = `
<div
  style="height: 500px; display: block;"
/>
`;
